### PR TITLE
Added support for linux/arm64 images in helm-charts

### LINKED ISSUE #330: cockroach-self-signer-cert:1.4 is not multi-arch
This currently fails to uninstall due to that fact.

### PR DESCRIPTION
**Pull request by prafull01**
Fixes #330

### CODE REVIEW

File: build/docker-image/self-signer-cert-utility/Dockerfile [hunk 1]
```diff
@@ -18,13 +18,16 @@ COPY cmd/ cmd/
...
 
 # Build the binary self-signer utility
-RUN go build -a -installsuffix cgo -o self-signer cmd/main.go
+RUN go build -o self-signer cmd/main.go
 
 # Install the cockroach CLI
-ADD https://binaries.cockroachdb.com/cockroach-v23.1.2.linux-amd64.tgz .
-RUN tar -zxvf cockroach-v23.1.2.linux-amd64.tgz
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then GOARCH=amd64; elif [ "$TARGETPLATFORM" = "linux/arm64" ]; then GOARCH=arm64; else GOARCH=amd64; fi && \
+    curl -sS -L -O https://binaries.cockroachdb.com/cockroach-v23.1.2.linux-${GOARCH}.tgz && \
```
Comment (rail) [inline build/docker-image/self-signer-cert-utility/Dockerfile:25]: A nit. Can you update the crdb version to the latest one? As a bonus you can move it to a variable which can be passed in. Not a blocker though, feel free to do this as a separate PR.

File: build/docker-image/self-signer-cert-utility/Dockerfile [hunk 2]
```diff
@@ -18,13 +18,16 @@ COPY cmd/ cmd/
 COPY pkg/ pkg/
 
 # Build the binary self-signer utility
-RUN go build -a -installsuffix cgo -o self-signer cmd/main.go
+RUN go build -o self-signer cmd/main.go
 
 # Install the cockroach CLI
```
Comment (rail) [inline build/docker-image/self-signer-cert-utility/Dockerfile:23]: A nit. "cockroach CLI" means `cockroach-cli`, which is a separate binary and doesn't contain any "server" code. Maybe adjust the comment to something like "Install the cockroach binary"?

**Review by rail — APPROVED** [review]

File: build/docker-image/self-signer-cert-utility/Dockerfile (hunk 1 above)
Comment (prafull01) [reply build/docker-image/self-signer-cert-utility/Dockerfile]: I did even better, now the self-signer automatically picks up the latest cockroach binary version defined in the Chart.yaml. Now, no need for special code changes to upgrade cockroach binary in docker image. Please have a look.

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -43,15 +43,18 @@ jobs:
...
         uses: actions/checkout@v2
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
       - name: Login to GCR
         uses: docker/login-action@v1
```
Comment (mprimeaux) [inline .github/workflows/build.yaml:53]: The latest version of this GitHub action is [v3](https://github.com/docker/login-action/releases/tag/v3.0.0). Do you prefer to upgrade from v1?

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -43,15 +43,18 @@ jobs:
       - name: Checkout sources
         uses: actions/checkout@v2
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
```
Comment (mprimeaux) [inline .github/workflows/build.yaml:47]: The latest version of this GitHub action is [v3](https://github.com/docker/setup-qemu-action/releases/tag/v3.0.0). Do you prefer to upgrade from v2?

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -43,15 +43,18 @@ jobs:
...
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
```
Comment (mprimeaux) [inline .github/workflows/build.yaml:50]: The latest version of this GitHub action is [v3](https://github.com/docker/setup-buildx-action/releases/tag/v3.0.0). Do you prefer to upgrade from v2?

File: build/docker-image/self-signer-cert-utility/Dockerfile [hunk 6]
```diff
@@ -18,13 +19,16 @@ COPY cmd/ cmd/
...
 
 # Build the binary self-signer utility
-RUN go build -a -installsuffix cgo -o self-signer cmd/main.go
+RUN go build -o self-signer cmd/main.go
 
-# Install the cockroach CLI
-ADD https://binaries.cockroachdb.com/cockroach-v23.1.2.linux-amd64.tgz .
-RUN tar -zxvf cockroach-v23.1.2.linux-amd64.tgz
+# Install the cockroach binary
+RUN if [ "$TARGETPLATFORM" = "linux/amd64" ]; then GOARCH=amd64; elif [ "$TARGETPLATFORM" = "linux/arm64" ]; then GOARCH=arm64; else GOARCH=amd64; fi && \
```
Comment (mprimeaux) [inline build/docker-image/self-signer-cert-utility/Dockerfile:25]: Just a minor point but if you were to add `TARGETOS` and `TARGETARCH` args then you can set `GOOS` and `GOARCH` to those values, respectively. Those args are passed in by `docker buildx` (and by `nerdctl build`) intrinsically.

I usually include the following block in my "build" stage so I see what's going on a bit more. I'm not suggesting this change but just putting it here for awareness. You can see where I set `GOOS` and `GOARCH`.

```dockerfile
ARG BUILDOS
ARG BUILDPLATFORM
ARG BUILDARCH
ARG BUILDVARIANT
ARG TARGETPLATFORM
ARG TARGETOS
ARG TARGETARCH
ARG TARGETVARIANT
ARG BUILD_VERSION
ARG BUILD_COMMIT

RUN echo "Container image build platform details :: "
RUN echo "BUILDPLATFORM.....: $BUILDPLATFORM"
RUN echo "BUILDOS...........: $BUILDOS"
RUN echo "BUILDARCH.........: $BUILDARCH"
RUN echo "BUILDVARIANT......: $BUILDVARIANT"
RUN echo "Container image target platform details :: "
RUN echo "TARGETPLATFORM....: $TARGETPLATFORM"
RUN echo "TARGETOS..........: $TARGETOS"
RUN echo "TARGETARCH........: $TARGETARCH"
RUN echo "TARGETVARIANT.....: $TARGETVARIANT"

ENV GOOS=$TARGETOS
ENV GOARCH=$TARGETARCH
```

File: cockroachdb/values.yaml [hunk 7]
```diff
@@ -519,6 +519,21 @@ tls:
...
+    affinity: {}
+
+    # Node selection constraints for scheduling the Pod of this Job.
+    # https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+    nodeSelector: {}
+
+    # Taints to be tolerated by the Pod of this Job.
+    # https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+    tolerations: []
+
```
Comment (mprimeaux) [inline cockroachdb/values.yaml:536]: Thanks for adding these!

**Review by mprimeaux — COMMENTED** [review]
Thanks for these changes!

File: .github/workflows/build.yaml (hunk 3 above)
Comment (prafull01) [reply .github/workflows/build.yaml:53]: Upgrade to v3 will be done in separate PR.

File: .github/workflows/build.yaml (hunk 4 above)
Comment (prafull01) [reply .github/workflows/build.yaml:47]: Upgrade to v3 will be done in separate PR and will be done in all github actions.

**Review by mprimeaux — COMMENTED** [review]
LGTM

**Pull request merged**